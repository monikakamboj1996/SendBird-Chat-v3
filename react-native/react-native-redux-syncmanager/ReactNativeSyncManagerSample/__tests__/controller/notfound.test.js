import 'react-native';
import React from 'react';
import NotFound from '../../src/controller/notfound';
import renderer from 'react-test-renderer';

const navigation = {
  setParams: jest.fn(),
  getParam: jest.fn().mockReturnValue('channel Url'),
  navigate: jest.fn(),
  state: {
    params: {
      channelUrl: 'channelUrl',
      isOpenChannel: true
    }
  },
  dispatch: jest.fn(),
  addListener: jest.fn()
};

describe('controller/notfound', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NotFound navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
