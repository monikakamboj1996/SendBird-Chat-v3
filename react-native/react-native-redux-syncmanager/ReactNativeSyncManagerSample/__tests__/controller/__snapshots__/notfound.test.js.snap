// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controller/notfound renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f3f3f3",
      "flex": 1,
      "justifyContent": "center",
      "padding": 40,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#999",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    The channel does not exist or you are not a member of this channel.
  </Text>
</View>
`;
